assets/js/utils/render-frontend.js

```diff
@@ -2,6 +2,44 @@
  * External dependencies
  */
 import { render } from 'react-dom';
+import { getSetting } from '@woocommerce/settings';
+import { SCHEMA_STORE_KEY } from '@woocommerce/block-data';
+import { useEffect, useRef } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { forEach } from 'lodash';
+
+/**
+ * Hydrate Rest API data from settings to reduce the number of API requests needed.
+ *
+ * @param {mixed} children Child components.
+ */
+const HydrateRestApiData = ( { children } ) => {
+	const { receiveRoutes, startResolution, finishResolution } = useDispatch(
+		SCHEMA_STORE_KEY
+	);
+	const { isResolving, hasFinishedResolution } = useSelect( ( select ) => {
+		const store = select( SCHEMA_STORE_KEY );
+		return store;
+	} );
```

Comment:
There's a couple problems with this approach:

- `useSelect` is wired so that whenever the store state changes, it invokes the  provided callback for retrieving data from the state _at that point of time_.  This works because of an internal subscription to the store set by `useSelect`.   When you just return the selectors themselves, you are only going to get the condition of the state at the render point for that component (which may or may not be what you want... it's a crapshoot).
- One way you might consider having worked around this issue is by not adding a dependency on the `useSelect` you are having it set every time the `HydrateRestApiData` goes through the lifecycles for a component.  This is potentially expensive because it essentially means the callback is invoked multiple times.